pkg/i2gw/providers/common/utils.go

```diff
@@ -226,3 +230,158 @@ func removeBackendRefsDuplicates(backendRefs []gatewayv1.HTTPBackendRef) []gatew
+		method = parts[1]
+	}
+
+	return service, method
+}
+
+// GRPCFilterConversionResult holds the result of converting HTTP filters to gRPC filters
+type GRPCFilterConversionResult struct {
+	GRPCFilters      []gatewayv1.GRPCRouteFilter
+	UnsupportedTypes []gatewayv1.HTTPRouteFilterType
+}
+
+// ConvertHTTPFiltersToGRPCFilters converts a list of HTTPRoute filters to GRPCRoute filters
+// Returns both the converted filters and a list of unsupported filter types for notification
+func ConvertHTTPFiltersToGRPCFilters(httpFilters []gatewayv1.HTTPRouteFilter) GRPCFilterConversionResult {
+	if len(httpFilters) == 0 {
+		return GRPCFilterConversionResult{
+			GRPCFilters:      []gatewayv1.GRPCRouteFilter{},
+			UnsupportedTypes: []gatewayv1.HTTPRouteFilterType{},
+		}
+	}
+
+	var grpcFilters []gatewayv1.GRPCRouteFilter
+	var unsupportedTypes []gatewayv1.HTTPRouteFilterType
+
+	for _, httpFilter := range httpFilters {
+		switch httpFilter.Type {
+		case gatewayv1.HTTPRouteFilterRequestHeaderModifier:
+			if httpFilter.RequestHeaderModifier != nil {
+				grpcFilter := gatewayv1.GRPCRouteFilter{
+					Type: gatewayv1.GRPCRouteFilterRequestHeaderModifier,
+					RequestHeaderModifier: &gatewayv1.HTTPHeaderFilter{
+						Set:    httpFilter.RequestHeaderModifier.Set,
+						Add:    httpFilter.RequestHeaderModifier.Add,
+						Remove: httpFilter.RequestHeaderModifier.Remove,
+					},
+				}
+				grpcFilters = append(grpcFilters, grpcFilter)
+			}
+
+		case gatewayv1.HTTPRouteFilterResponseHeaderModifier:
+			if httpFilter.ResponseHeaderModifier != nil {
+				grpcFilter := gatewayv1.GRPCRouteFilter{
+					Type: gatewayv1.GRPCRouteFilterResponseHeaderModifier,
+					ResponseHeaderModifier: &gatewayv1.HTTPHeaderFilter{
+						Set:    httpFilter.ResponseHeaderModifier.Set,
+						Add:    httpFilter.ResponseHeaderModifier.Add,
+						Remove: httpFilter.ResponseHeaderModifier.Remove,
+					},
+				}
+				grpcFilters = append(grpcFilters, grpcFilter)
+			}
+
+		// These HTTP filter types are not applicable to gRPC
+		case gatewayv1.HTTPRouteFilterRequestRedirect:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		case gatewayv1.HTTPRouteFilterURLRewrite:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		case gatewayv1.HTTPRouteFilterRequestMirror:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		case gatewayv1.HTTPRouteFilterExtensionRef:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		default:
+			unsupportedTypes = append(unsupportedTypes, httpFilter.Type)
+		}
+	}
+
+	// Ensure we return empty slices instead of nil
+	if grpcFilters == nil {
+		grpcFilters = []gatewayv1.GRPCRouteFilter{}
+	}
+	if unsupportedTypes == nil {
+		unsupportedTypes = []gatewayv1.HTTPRouteFilterType{}
+	}
+
+	return GRPCFilterConversionResult{
+		GRPCFilters:      grpcFilters,
+		UnsupportedTypes: unsupportedTypes,
+	}
+}
+
+// RemoveGRPCRulesFromHTTPRoute removes HTTPRoute rules that target gRPC services
```

Comment:
why is this function needed? I see where it is used. will be useful to elaborate on this in the comment